src/AzSK.ADO/Framework/Abstracts/MetaInfoProvider.ps1

```diff
@@ -0,0 +1,339 @@
+                    $rsrcList = $this.CallADOInfoAPI($qs);
+                    $apiReturnedResourceTypeName = $resourceTypeName+"s";
+                    if ($rsrcList -and ( [Helpers]::CheckMember($rsrcList, "$apiReturnedResourceTypeName") -and $rsrcList."$apiReturnedResourceTypeName") ) {
+                        $this.BindADOInfoAPIResponseToSTMappingFiles($rsrcList, $resourceTypeName);
+                    }
+                    #If not get files from adoinfoapi, take then from local org policy files. 
+                    #else {
+                    #    $this.FetchMappingFiles($resourceTypeName);
+                    #}
+                }
+            }
+            else 
+            {
+                $this.FetchMappingFiles($resourceTypeName);
+            }
+
+            $serviceTreeInfo = $this.GetServiceDataForResource($rscId, $resourceTypeName);
+        }
+        catch
+        {
+            Write-Host "Could not fetch service mapping files. `r`nPlease contact your project's ADO security team." -ForegroundColor Red 
+        }
+        return $serviceTreeInfo; 
+    }
+
+    #Binding adoinfo api response to class local variable
+    hidden [void] BindADOInfoAPIResponseToSTMappingFiles($resourceList, $resourceTypeName)
+    {
+        if ($resourceTypeName -eq "Build") {
+            $this.buildSTDetails = $resourceList.Builds;
+        }
+        elseif ($resourceTypeName -eq "Release") {
+            $this.releaseSTDetails = $resourceList.Releases;
+        }
+        elseif ($resourceTypeName -eq "ServiceConnection") {
+            $this.svcConnSTDetails = $resourceList.ServiceConnections;
+        }
+        elseif ($resourceTypeName -eq "AgentPool") {
+            $this.agtPoolSTDetails = $resourceList.AgentPools;
+        }
+        elseif ($resourceTypeName -eq "VariableGroup") {
+            $this.varGroupSTDetails = $resourceList.VariableGroups;
+        }
+        elseif ($resourceTypeName -eq "ServiceTree") {
+            $this.serviceTreeDetails = $resourceList.ServiceTree;
+        }
+    }
+
+    #Loading local org policy ST files based on supplied resource type, 
+    #1.Fetch ST files from policy only if ...STDetails variable is null (if not already fetch)
+    #2.Do not fetch ST files again from policy, if already fetched and file is not present in policy server.
+    [void] FetchMappingFiles($ResourceTypeName)
+	{
+		if ($ResourceTypeName -in ([ResourceTypeName]::Build, [ResourceTypeName]::All, [ResourceTypeName]::Build_Release, [ResourceTypeName]::Build_Release_SvcConn_AgentPool_User))
+		{
+		   if (!$this.buildSTDetails -and $this.checkBuildSTFileOnServer) {
+                $this.buildSTDetails = [ConfigurationManager]::LoadServerConfigFile("BuildSTData.json");
+                #If file is fetch and return null from policy server
+                if (!$this.buildSTDetails) {
```

Comment:
remove this if clause. checkBuildSTFileOnServer is one time computation irrespective of whether it is found or not